app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -38,16 +58,60 @@ class MainActivity : ComponentActivity() {
     @Inject
     lateinit var lazyStats: dagger.Lazy<JankStats>
 
+    @Inject
+    lateinit var userDataRepository: UserDataRepository
+
     override fun onCreate(savedInstanceState: Bundle?) {
-        installSplashScreen()
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
+        /**
+         * The current user data, updated here to drive the UI theme
+         */
+        var userDataResult: Result<UserData> by mutableStateOf(Result.Loading)
+
+        // Update the user data
+        lifecycleScope.launch {
```

Comment:
I'd move any UI logic to a VM, even if it's super simple. Just collect the userDataResult from it here.